src/main/java/com/gluonhq/substrate/target/AndroidTargetConfiguration.java

```diff
@@ -64,37 +65,45 @@
 
 public class AndroidTargetConfiguration extends PosixTargetConfiguration {
 
+    private static final String ANDROID_TRIPLET = "aarch64-linux-android";
+    private static final String ANDROID_MIN_SDK_VERSION = "21";
+
     private final String ndk;
     private final String sdk;
     private final Path ldlld;
     private final Path clang;
+    private final Path objdump;
     private final String hostPlatformFolder;
 
-    private List<String> androidAdditionalSourceFiles = Arrays.asList("launcher.c", "javafx_adapter.c",
+    private final List<String> androidAdditionalSourceFiles = Arrays.asList("launcher.c", "javafx_adapter.c",
             "touch_events.c", "glibc_shim.c", "attach_adapter.c", "logger.c");
-    private List<String> androidAdditionalHeaderFiles = Arrays.asList("grandroid.h", "grandroid_ext.h");
-    private List<String> cFlags = Arrays.asList("-target", "aarch64-linux-android", "-I.");
-    private List<String> linkFlags = Arrays.asList("-target", "aarch64-linux-android21", "-fPIC", "-fuse-ld=gold",
+    private final List<String> androidAdditionalHeaderFiles = Arrays.asList("grandroid.h", "grandroid_ext.h");
+    private final List<String> cFlags = Arrays.asList("-target", ANDROID_TRIPLET, "-I.");
+    private final List<String> linkFlags = Arrays.asList("-target",
+            ANDROID_TRIPLET + ANDROID_MIN_SDK_VERSION, "-fPIC", "-fuse-ld=gold",
             "-Wl,--rosegment,--gc-sections,-z,noexecstack", "-shared",
             "-landroid", "-llog", "-lffi", "-llibchelper");
-    private List<String> javafxLinkFlags = Arrays.asList("-Wl,--whole-archive",
+    private final List<String> javafxLinkFlags = Arrays.asList("-Wl,--whole-archive",
             "-lprism_es2_monocle", "-lglass_monocle", "-ljavafx_font_freetype", "-ljavafx_iio", "-Wl,--no-whole-archive",
             "-lGLESv2", "-lEGL", "-lfreetype");
     private final String capLocation = ANDROID_NATIVE_FOLDER + "cap/";
 
-    public AndroidTargetConfiguration( ProcessPaths paths, InternalProjectConfiguration configuration ) throws IOException {
+    public AndroidTargetConfiguration(ProcessPaths paths, InternalProjectConfiguration configuration) throws IOException {
         super(paths,configuration);
 
         this.sdk = fileDeps.getAndroidSDKPath().toString();
         this.ndk = fileDeps.getAndroidNDKPath().toString();
-        this.hostPlatformFolder = configuration.getHostTriplet().getOs() + "-x86_64";
+        this.hostPlatformFolder = configuration.getHostTriplet().getOs() + "-" + Constants.ARCH_AMD64;
```

Comment:
isn't this Triplet.getOsArch() ?

Reply:
No, that will return the Android (target) value, not the host one.